circuit_definitions/src/circuit_definitions/base_layer/ecadd.rs

```diff
@@ -0,0 +1,178 @@
+use crate::zkevm_circuits::bn254::ec_add::input::EcAddCircuitInstanceWitness;
+use circuit_encodings::zkevm_circuits::bn254::{
+    ec_add::ecadd_function_entry_point,
+    fixed_base_mul_table::{create_fixed_base_mul_table, FixedBaseMulTable},
+};
+use derivative::*;
+
+use super::*;
+use crate::boojum::cs::traits::circuit::CircuitBuilder;
+
+type F = GoldilocksField;
+type R = Poseidon2Goldilocks;
+
+#[derive(Derivative, serde::Serialize, serde::Deserialize)]
+#[derivative(Clone, Copy, Debug, Default(bound = ""))]
+pub struct ECAddFunctionInstanceSynthesisFunction {
+    _marker: std::marker::PhantomData<(F, R)>,
+}
+
+impl CircuitBuilder<F> for ECAddFunctionInstanceSynthesisFunction
+where
+    [(); <LogQuery<F> as CSAllocatableExt<F>>::INTERNAL_STRUCT_LEN]:,
+    [(); <MemoryQuery<F> as CSAllocatableExt<F>>::INTERNAL_STRUCT_LEN]:,
+    [(); <DecommitQuery<F> as CSAllocatableExt<F>>::INTERNAL_STRUCT_LEN]:,
+    [(); <UInt256<F> as CSAllocatableExt<F>>::INTERNAL_STRUCT_LEN]:,
+    [(); <UInt256<F> as CSAllocatableExt<F>>::INTERNAL_STRUCT_LEN + 1]:,
+{
+    fn geometry() -> CSGeometry {
+        CSGeometry {
+            num_columns_under_copy_permutation: 200,
+            num_witness_columns: 0,
+            num_constant_columns: 8,
+            max_allowed_constraint_degree: 4,
+        }
+    }
+
+    fn lookup_parameters() -> LookupParameters {
+        LookupParameters::UseSpecializedColumnsWithTableIdAsConstant {
+            width: 3,
+            num_repetitions: 8,
+            share_table_id: true,
+        }
+    }
+
+    fn configure_builder<
+        T: CsBuilderImpl<F, T>,
+        GC: GateConfigurationHolder<F>,
+        TB: StaticToolboxHolder,
+    >(
+        builder: CsBuilder<T, F, GC, TB>,
+    ) -> CsBuilder<T, F, impl GateConfigurationHolder<F>, impl StaticToolboxHolder> {
+        // TODO: Maybe some gates are not actually needed since it was copy-pasting from the previous secp256k1 ecmul implementation.
```

Comment:
you can check if this is the case or not by running a full proof of the circuit and then printing the gate stats - it should give you a tally of how often each gate is called